tools/rtpsrelay/RelayHttpMetaDiscovery.cpp

```diff
@@ -0,0 +1,146 @@
+#include "RelayHttpMetaDiscovery.h"
+
+#include <dds/DCPS/LogAddr.h>
+#include <dds/DCPS/Service_Participant.h>
+#include <dds/DCPS/ThreadStatusManager.h>
+
+#include <ace/INET_Addr.h>
+
+#include <cstdlib>
+#include <cstring>
+#include <sstream>
+#include <string>
+
+namespace RtpsRelay {
+
+const HttpStatus HTTP_OK(200, "OK");
+const HttpStatus HTTP_NOT_FOUND(404, "Not Found");
+const HttpStatus HTTP_SERVICE_UNAVAILABLE(503, "Service Unavailable");
+
+const int HANDLER_ERROR = -1, HANDLER_REMOVE = -1, HANDLER_OK = 0;
+
+int HttpConnection::open(void* x)
+{
+  relay_http_meta_discovery_ = static_cast<RelayHttpMetaDiscovery*>(x);
+
+  if (-1 == reactor()->register_handler(this, READ_MASK)) {
+    return HANDLER_ERROR;
+  }
+  return HANDLER_OK;
+}
+
+int HttpConnection::handle_input(ACE_HANDLE)
+{
+  OpenDDS::DCPS::ThreadStatusManager::Event ev(TheServiceParticipant->get_thread_status_manager());
+
+  const auto bytes = peer().recv(buffer_.wr_ptr(), buffer_.space(),
+                                 &ACE_Time_Value::zero);
+
+  if (bytes == 0) {
+    peer().close_reader();
+    return HANDLER_REMOVE;
+  }
+
+  if (bytes < 0 && errno == ETIME) {
+    return HANDLER_OK;
+  }
+
+  if (bytes <= 0) {
+    return HANDLER_ERROR;
+  }
+
+  buffer_.wr_ptr(bytes);
+  const std::string request(buffer_.rd_ptr(), buffer_.length());
+  if (relay_http_meta_discovery_->processRequest(peer(), request)) {
+    peer().close_writer();
+    buffer_.reset();
+  }
+
+  return HANDLER_OK;
+}
+
+bool RelayHttpMetaDiscovery::processRequest(ACE_SOCK_Stream& peer,
+                                            const std::string& request) const
+{
+  std::string target;
+  if (requestIsComplete(request, target)) {
+    std::stringstream response;
+    if (target == "/") {
+      respond(response);
```

Comment:
Why not use `relay.json` or something similar to name the resource?  It seems like that could better support extensibility.